integration-tests/actions/vrf/vrfv2plus/contract_steps.go

```diff
@@ -0,0 +1,770 @@
+			*config.MinimumConfirmations,
+			*config.CallbackGasLimit,
+			*config.NumberOfWords,
+			*config.RandomnessRequestCountPerRequest,
+		)
+		if err != nil {
+			return "", fmt.Errorf("%s, err %w", ErrRequestRandomnessDirectFundingNativePayment, err)
+		}
+	} else {
+		_, err := consumer.RequestRandomness(
+			*config.MinimumConfirmations,
+			*config.CallbackGasLimit,
+			*config.NumberOfWords,
+			*config.RandomnessRequestCountPerRequest,
+		)
+		if err != nil {
+			return "", fmt.Errorf("%s, err %w", ErrRequestRandomnessDirectFundingLinkPayment, err)
+		}
+	}
+	wrapperAddress, err := consumer.GetWrapper(context.Background())
+	if err != nil {
+		return "", fmt.Errorf("error getting wrapper address, err: %w", err)
+	}
+	return wrapperAddress.Hex(), nil
+}
+
+func DirectFundingRequestRandomnessAndWaitForFulfillment(
+	consumer contracts.VRFv2PlusWrapperLoadTestConsumer,
+	coordinator contracts.VRFCoordinatorV2_5,
+	vrfKeyData *vrfcommon.VRFKeyData,
+	subID *big.Int,
+	isNativeBilling bool,
+	config *vrfv2plus_config.General,
+	l zerolog.Logger,
+) (*vrf_coordinator_v2_5.VRFCoordinatorV25RandomWordsFulfilled, error) {
+	wrapperAddress, err := WrapperRequestRandomness(consumer, coordinator.Address(), vrfKeyData, subID,
+		isNativeBilling, config, l)
+	if err != nil {
+		return nil, fmt.Errorf("error getting wrapper address, err: %w", err)
+	}
+	return WaitForRequestAndFulfillmentEvents(
+		wrapperAddress,
+		coordinator,
+		vrfKeyData,
+		subID,
+		isNativeBilling,
+		config.RandomWordsFulfilledEventTimeout.Duration,
+		l,
+	)
+}
+
+func DirectFundingRequestRandomnessAndWaitForFulfillmentUpgraded(
+	consumer contracts.VRFv2PlusWrapperLoadTestConsumer,
+	coordinator contracts.VRFCoordinatorV2PlusUpgradedVersion,
+	vrfKeyData *vrfcommon.VRFKeyData,
+	subID *big.Int,
+	isNativeBilling bool,
+	config *vrfv2plus_config.General,
+	l zerolog.Logger,
+) (*vrf_v2plus_upgraded_version.VRFCoordinatorV2PlusUpgradedVersionRandomWordsFulfilled, error) {
+	wrapperAddress, err := WrapperRequestRandomness(consumer, coordinator.Address(), vrfKeyData, subID,
+		isNativeBilling, config, l)
+	if err != nil {
+		return nil, fmt.Errorf("error getting wrapper address, err: %w", err)
+	}
+	return WaitForRequestAndFulfillmentEventsUpgraded(
+		wrapperAddress,
+		coordinator,
+		vrfKeyData,
+		subID,
+		isNativeBilling,
+		config.RandomWordsFulfilledEventTimeout.Duration,
+		l,
+	)
+}
+
+func WaitForRequestAndFulfillmentEvents(
+	consumerAddress string,
+	coordinator contracts.VRFCoordinatorV2_5,
+	vrfKeyData *vrfcommon.VRFKeyData,
+	subID *big.Int,
+	isNativeBilling bool,
+	randomWordsFulfilledEventTimeout time.Duration,
+	l zerolog.Logger,
+) (*vrf_coordinator_v2_5.VRFCoordinatorV25RandomWordsFulfilled, error) {
+	randomWordsRequestedEvent, err := coordinator.WaitForRandomWordsRequestedEvent(
+		[][32]byte{vrfKeyData.KeyHash},
+		[]*big.Int{subID},
+		[]common.Address{common.HexToAddress(consumerAddress)},
+		time.Minute*1,
```

Comment:
But `time.Minute*1` is equal to `time.Minute`? 🤔 This code repeats in several places so I don't if it's intentional for readability or...?

Reply:
just for readability.

My intention was to tackle all the timeouts as they should be parametrized and set depending on the chain. part of https://smartcontract-it.atlassian.net/browse/VRF-940